### PR TITLE
Management Plane API Switch

### LINKED ISSUE #1069: Switch over to azure-sdk-for-js
https://github.com/Azure/azure-sdk-for-js/blob/master/documentation/Migration.md

(This is for management plan - https://github.com/Microsoft/vscode-cosmosdb/issues/760 covers data plan)

### PR DESCRIPTION
**Pull request by PrashanthCorp**
azure-arm-xyz to @azure/arm-xyz

Fixes #1069 

Builds will fail till the corresponding PR in azuretools is merged and published: https://github.com/microsoft/vscode-azuretools/pull/511

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -905,22 +905,25 @@
...
+		"html-to-text": "^4.0.0",
 		"mongodb": "^2.2.25",
 		"mongodb-extended-json": "^1.10.0",
-		"ms-rest": "^2.2.1",
 		"node-uuid": "1.4.8",
 		"socket.io": "^1.7.3",
 		"socket.io-client": "^1.7.3",
 		"underscore": "^1.8.3",
-		"vscode-azureextensionui": "^0.24.0",
+		"vscode-azureextensionui": "file:../vscode-azuretools/ui/vscode-azureextensionui-0.25.0.tgz",
```
Comment (PrashanthCorp) [inline package.json]: Will change this to 0.25.0 once this PR is merged: https://github.com/microsoft/vscode-azuretools/pull/511

File: src/docdb/tree/DocDBAccountTreeItemBase.ts [hunk 2]
```diff
@@ -3,7 +3,7 @@
  *  Licensed under the MIT License. See License.txt in the project root for license information.
  *--------------------------------------------------------------------------------------------*/
 
-import { DatabaseAccount } from 'azure-arm-cosmosdb/lib/models';
+import { DatabaseAccount } from '@azure/arm-cosmosdb/esm/models';
```
Comment (EricJizbaMSFT) [inline src/docdb/tree/DocDBAccountTreeItemBase.ts]: Please switch these to use the `Models` object from the base of the npm package, which is more of an npm package's "official" api. Something like this

```
import { ResourceManagementModels } from '@azure/arm-resources';
```

File: package.json [hunk 3]
```diff
@@ -905,22 +905,25 @@
 		"webpack-cli": "^3.1.2"
 	},
 	"dependencies": {
+		"@azure/arm-cosmosdb": "^3.0.0",
+		"@azure/ms-rest-azure-js": "^1.3.5",
```
Comment (EricJizbaMSFT) [inline package.json]: Remove the node modules that you're not using

File: package.json [hunk 4]
```diff
@@ -905,22 +905,25 @@
...
 		"gremlin": "^2.6.0",
+		"html-to-text": "^4.0.0",
 		"mongodb": "^2.2.25",
 		"mongodb-extended-json": "^1.10.0",
-		"ms-rest": "^2.2.1",
 		"node-uuid": "1.4.8",
 		"socket.io": "^1.7.3",
 		"socket.io-client": "^1.7.3",
 		"underscore": "^1.8.3",
-		"vscode-azureextensionui": "^0.24.0",
```
Comment (EricJizbaMSFT) [inline package.json]: This is out of date - merge with master

File: src/docdb/tree/DocDBAccountTreeItemBase.ts (hunk 2 above)
Comment (PrashanthCorp) [reply src/docdb/tree/DocDBAccountTreeItemBase.ts]: Is this a newer preference? This has been how it's used everywhere. 
For eg. [appservice](https://github.com/microsoft/vscode-azureappservice/blob/25cb5fd10fe62d5992626826858bc10d70677904/src/explorer/DeploymentSlotsTreeItem.ts#L7)

Comment (EricJizbaMSFT) [reply src/docdb/tree/DocDBAccountTreeItemBase.ts]: I mean it's not super new, but yeah it changed sometime after we originally created the extensions. For example the original app service extension referenced v1.0.0-preview of "azure-arm-website" which didn't even have the models object:
Screen Shot 2019-06-21 at 4 08 42 PM

Now we're on a much later version that does:
Screen Shot 2019-06-21 at 4 09 43 PM

Functions is already moved over: https://github.com/microsoft/vscode-azurefunctions/blob/795ab870b56e133e2c360b3bf08df9419b792d4a/src/tree/FunctionsTreeItem.ts#L6
Docker is moved over: https://github.com/microsoft/vscode-docker/blob/2fae7a525c41f6f8375f51fc6a3598e2d05e5ca4/src/tree/registries/azure/AzureTaskRunTreeItem.ts#L6

There's a tslint rule here that we can eventually turn on: https://palantir.github.io/tslint/rules/no-submodule-imports/

Comment (PrashanthCorp) [reply src/docdb/tree/DocDBAccountTreeItemBase.ts]: Thanks for the heads up. 

I turned on the rule, it looks like there are other places where the rule is broken. Leaving those at the moment.

**Pull request closed without merging**